commons-numbers-gamma/src/main/java/org/apache/commons/numbers/gamma/RegularizedGamma.java

```diff
@@ -0,0 +1,210 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.numbers.gamma;
+
+import java.text.MessageFormat;
+
+import org.apache.commons.numbers.fraction.ContinuedFraction;
+
+/**
+ * <a href="http://mathworld.wolfram.com/RegularizedGammaFunction.html">
+ * Regularized Gamma functions</a>.
+ *
+ * Class is immutable.
+ */
+public final class RegularizedGamma {
+    /** Maximum allowed numerical error. */
+    private static final double DEFAULT_EPSILON = 1e-15;
+
+    /** Private constructor. */
+    private RegularizedGamma() {
+        // intentionally empty.
+    }
+
+    /**
+     * \( P(a, x) \) <a href="http://mathworld.wolfram.com/RegularizedGammaFunction.html">
+     * regularized Gamma function</a>.
+     *
+     * Class is immutable.
+     */
+    public static final class P {
+        /** Prevent instantiation. */
+        private P() {}
+
+        /**
+         * Computes the regularized gamma function \( P(a, x) \).
+         *
+         * @param a Argument.
+         * @param x Argument.
+         * @return \( P(a, x) \).
+         * @throws ArithmeticException if the continued fraction fails to converge.
+         */
+        public static double value(double a,
+                                   double x) {
+            return value(a, x, DEFAULT_EPSILON, Integer.MAX_VALUE);
+        }
+
+        /**
+         * Computes the regularized gamma function \( P(a, x) \).
+         *
+         * The implementation of this method is based on:
+         * <ul>
+         *  <li>
+         *   <a href="http://mathworld.wolfram.com/RegularizedGammaFunction.html">
+         *   Regularized Gamma Function</a>, equation (1)
+         *  </li>
+         *  <li>
+         *   <a href="http://mathworld.wolfram.com/IncompleteGammaFunction.html">
+         *   Incomplete Gamma Function</a>, equation (4).
+         *  </li>
+         *  <li>
+         *   <a href="http://mathworld.wolfram.com/ConfluentHypergeometricFunctionoftheFirstKind.html">
+         *   Confluent Hypergeometric Function of the First Kind</a>, equation (1).
+         *  </li>
+         * </ul>
+         *
+         * @param a Argument.
+         * @param x Argument.
+         * @param epsilon Tolerance in continued fraction evaluation.
+         * @param maxIterations Maximum number of iterations in continued fraction evaluation.
+         * @return \( P(a, x) \).
+         * @throws ArithmeticException if the continued fraction fails to converge.
+         */
+        public static double value(double a,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.